src/core/init/utils/initial_seek_and_play.ts

```diff
@@ -127,7 +127,47 @@ export default function performInitialSeekAndPlay(
     // a sufficient `readyState` has been reached for directfile contents.
     // So let's divide the two possibilities here.
     const initialTime = typeof startTime === "function" ? startTime() : startTime;
-    if (shouldPreventSeekingAt0Initially() && initialTime === 0) {
+    if (
+      initialTime === undefined &&
+      isDirectfile &&
+      mediaElement.readyState < HTMLMediaElement.HAVE_CURRENT_DATA
+    ) {
+      /**
+       * On browser, such as Safari, the HTMLMediaElement.duration
+       * and HTMLMediaElement.buffered may not be initialized at readyState 1, leading
+       * to cases where it can be equal to `Infinity`.
```

Comment:
Where what can be equal to `Infinity`? 
You're talking about `buffered`, but isn't it `duration` here? And isn't `duration` set to something like `NaN` by default.

Reply:
You are right the comment is not clear.
`duration` is either set to `NaN` or `Infinity`

> The duration attribute must return the time of the end of the [media resource](https://html.spec.whatwg.org/multipage/media.html#media-resource), in seconds, on the [media timeline](https://html.spec.whatwg.org/multipage/media.html#media-timeline). If no [media data](https://html.spec.whatwg.org/multipage/media.html#media-data) is available, then the attributes must return the Not-a-Number (NaN) value. If the [media resource](https://html.spec.whatwg.org/multipage/media.html#media-resource) is not known to be bounded (e.g. streaming radio, or a live event with no announced end time), then the attribute must return the positive Infinity value.